packages/platforms/browser/lib/auto-instrumentation/network-request-plugin.ts

```diff
@@ -59,15 +69,49 @@ export class NetworkRequestPlugin implements Plugin<BrowserConfiguration> {
     span.setAttribute('http.method', startContext.method)
     span.setAttribute('http.url', networkRequestInfo.url)
 
-    return (endContext: RequestEndContext) => {
-      if (endContext.state === 'success') {
-        span.setAttribute('http.status_code', endContext.status)
-        this.spanFactory.endSpan(span, endContext.endTime)
-      }
+    return {
+      onRequestEnd: (endContext: RequestEndContext) => {
+        if (endContext.state === 'success') {
+          span.setAttribute('http.status_code', endContext.status)
+          this.spanFactory.endSpan(span, endContext.endTime)
+        }
+      },
+      extraRequestHeaders: this.getExtraRequestHeaders(
+        startContext,
+        span
+      )
     }
   }
 
   private shouldTrackRequest (startContext: RequestStartContext): boolean {
     return startContext.url !== this.configEndpoint && permittedPrefixes.some((prefix) => startContext.url.startsWith(prefix))
   }
+
+  private getExtraRequestHeaders (startContext: RequestStartContext, span?: SpanInternal): Record<string, string> {
+    const extraRequestHeaders: Record<string, string> = {}
+
+    if (this.tracePropagationUrls.some(regexp => regexp.test(startContext.url))) {
+      if (span && span.parentSpanId) {
+        const traceId = span.traceId
+        const parentSpanId = span.parentSpanId
```

Comment:
It should be set to `span.id`.  It will be the _parent_ of the span created on the server.

Reply:
Okay, I was wondering why the `traceparent` had the same value for both with and without a network span in the unit tests, this explains it. fixed in f7ded53